pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_status.go

```diff
@@ -59,7 +68,70 @@ func gatewayAddresses(gateway *gatewayapi.Gateway, instance *mesh_k8s.GatewayIns
 	return addrs
 }
 
-func mergeGatewayListenerStatuses(gateway *gatewayapi.Gateway, conditions ListenerConditions) []gatewayapi.ListenerStatus {
+const everyListener = gatewayapi.SectionName("")
+
+type attachedRoutes struct {
+	// num is the number of allowed routes for a listener
+	num int32
+	// invalidRoute can be empty
+	invalidRoute kube_types.NamespacedName
```

Comment:
we only ever show the latest version?

Reply:
Hmm no I can include all of them actually.

Reply:
@lahabana I updated the PR, is that what you meant?

Reply:
Yep thx!